rclcpp/include/rclcpp/executor.hpp

```diff
@@ -33,6 +33,8 @@ namespace rclcpp
 namespace executor
 {
 
+enum executor_handle_t {subscriber_handle, service_handle, client_handle, guard_cond_handle};
```

Comment:
You could use the new C++11 enum types if you were so inclined:

http://www.cprogramming.com/c++11/c++11-nullptr-strongly-typed-enum-class.html